include/zephyr/arch/arm64/exc.h

```diff
@@ -47,6 +47,10 @@ struct __esf {
 	uint64_t lr;
 	uint64_t spsr;
 	uint64_t elr;
+#ifdef CONFIG_ARM64_ENABLE_FRAME_POINTER
+	uint64_t fp;
+	uint64_t _dummy;
```

Comment:
What's the reasoning for the extra `_dummy` value here?

Reply:
For AArch64 `sp` must be 16-byte aligned.

Reply:
But then you need to declar this struct as `__packed` ? adding alignment dummy variables to a non-packed structure is meaningless since the compiler can pad however it wants in that case?

Reply:
Two things:

1. We already have `__aligned(16);` for the ESF, and that should pretty much guarantee that the struct is correctly packed and padded since we are only using `uint64_t` registers in the struct. Practically the ESF size is padded to the closest 16-byte multiple by padding at the end of the struct if needed.

2. Given the previous point, and assumed that the ESF size is always 16-byte padded, technically the `_dummy` register here is not actually needed because in the assembly code we are a bit smart and we move the `sp` before accessing the variables, see: https://github.com/zephyrproject-rtos/zephyr/blob/c41958aa2d58c6b6939aeced8b21146fbecc14fb/arch/arm64/core/vector_table.S#L40
that means that we never really have a misaligned `sp` as long as the ESF size is a multiple of 16-bytes.

The `_dummy` register is a bit of a legacy habit I have from when the code needs to store something in the ESF and moving the `sp` at the same time, for example this is forbidden in AArch64:
```
str x29, [sp, #-8]!
```
because it leaves the `sp` misaligned, so you add `_dummy` variable that allows you to do:
```
stp x29, xzr, [sp, #-16]!
```

Anyway, since the `_dummy` variable is indeed useless in this case, I'm going to remove it. Thanks for flagging it.

Reply:
> 1. We already have `__aligned(16);` for the ESF, and that should pretty much guarantee that the struct is correctly packed and padded since we are only using `uint64_t` registers in the struct. Practically the ESF size is padded to the closest 16-byte multiple by padding at the end of the struct if needed.

The problem is that the C standard makes no guarantees as how much padding the compiler can insert in a non-packed structure. So the compiler could arbitrarily decide to inject an extra 64-bit padding for whatever reason.

Reply:
Sorry, maybe I was not clear enough.

If the compiler would decide to inject an extra 64-bit padding for whatever reason in wherever position that is fine **as long as** the size of the ESF is still a multiple of 16-bytes (that would be a waste of space of course but not a functional bug).

Now, if I'm reading the manual correctly, for that `__aligned(16)` is what we actually need but adding `packed` should guarantee that we don't have the extra padding allocation either, so I'm going to add that as well, it doesn't harm and prevents waste of space.

TBH I see ading a `BUILD_ASSERT` a nuisance more than anything else given that we are not going to have anything else than `uint64_t` in the ESF and we are not really interested in the actual size of the ESF but more in its alignment, and also considering how the actual size is depending on the `CONFIG_*` selected.